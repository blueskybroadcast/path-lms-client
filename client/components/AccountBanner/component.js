import React from 'react';
import PropTypes from 'prop-types';

const AccountBanner = ({ bannerUrl, bannerRedirectUrl, bannerAltText }) => {
  if (bannerUrl) {
    return (
      <section className="account-banner">
        <div className="wrapper">
          <a
            href={bannerRedirectUrl}
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              alt={bannerAltText}
              src={bannerUrl}
            />
          </a>
        </div>
      </section>
    );
  }
  return null;
};

AccountBanner.propTypes = {
  bannerUrl: PropTypes.string,
  bannerRedirectUrl: PropTypes.string,
  bannerAltText: PropTypes.string
};

AccountBanner.defaultProps = {
  bannerUrl: null,
  bannerRedirectUrl: null,
  bannerAltText: null
};

export default AccountBanner;
