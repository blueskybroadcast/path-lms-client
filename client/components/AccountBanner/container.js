import { connect } from 'react-redux';

import AccountBanner from './component';

import { accountBannerSelector } from '../../selectors/coursesSelectors';

const mapStateToProps = (state) => {
  const { bannerUrl, bannerRedirectUrl, bannerAltText } = accountBannerSelector(state);
  return { bannerUrl, bannerRedirectUrl, bannerAltText };
};

export default connect(mapStateToProps)(AccountBanner);
