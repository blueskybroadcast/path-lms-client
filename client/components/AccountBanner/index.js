import AccountBanner from './container';

export default AccountBanner;
