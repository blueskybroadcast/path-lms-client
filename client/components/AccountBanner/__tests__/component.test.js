import React from 'react';
import { shallow } from 'enzyme';
import AccountBanner from '../component';

test('should render AccountBanner correctly', () => {
  const props = {
    bannerUrl: 'banner url',
    bannerRedirectUrl: 'banner redirect url',
    bannerAltText: 'Banner Alt Text'
  };
  const wrapper = shallow(<AccountBanner {...props} />);
  expect(wrapper).toMatchSnapshot();
});
