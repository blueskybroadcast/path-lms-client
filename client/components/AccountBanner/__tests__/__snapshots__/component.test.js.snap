// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AccountBanner correctly 1`] = `
<section
  className="account-banner"
>
  <div
    className="wrapper"
  >
    <a
      href="banner redirect url"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Banner Alt Text"
        src="banner url"
      />
    </a>
  </div>
</section>
`;
